language/move-prover/tests/sources/functional/loop_unroll.move

```diff
@@ -0,0 +1,105 @@
+module 0x42::loop_unroll {
+    fun t1_success(): u64 {
+        let i = 0;
+        while ({
+            spec {
+                invariant [unroll = 3] true;
+            };
+            (i < 5)
+        }) {
+            i = i + 1;
+        };
+        i
+    }
+    spec t1_success {
+        // this is expecteds, as the loop never terminutes within 3 unrolling
+        ensures false;
+    }
+
+    fun t2_success(): u64 {
+        let i = 0;
+        while ({
+            // it is okay to unroll more than the loop iterations, it won't
+            // affect the correctness
+            spec {
+                invariant [unroll = 6] true;
```

Comment:
Together with the predicate `true`, this doesn't read very intuitive, but I guess we can improve this to the point where no spec blocks need to be attached to get the unrolling.